elements/lisk-transactions/src/13_vote_transaction.ts

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { intToBuffer } from '@liskhq/lisk-cryptography';
+import { isNumberString, validator } from '@liskhq/lisk-validator';
+
+import {
+	BaseTransaction,
+	StateStore,
+	StateStorePrepare,
+} from './base_transaction';
+import { MAX_INT64 } from './constants';
+import { convertToAssetError, TransactionError } from './errors';
+import { TransactionJSON } from './transaction_types';
+import { sortUnlocking } from './utils';
+
+export interface Vote {
+	readonly delegateAddress: string;
+	readonly amount: bigint;
+}
+
+export interface VoteAsset {
+	readonly votes: ReadonlyArray<Vote>;
+}
+
+const voteAssetFormatSchema = {
+	type: 'object',
+	required: ['votes'],
+	properties: {
+		votes: {
+			type: 'array',
+			minItems: 1,
+			maxItems: 20,
+			items: {
+				type: 'object',
+				required: ['delegateAddress', 'amount'],
+				properties: {
+					delegateAddress: {
+						type: 'string',
+						format: 'address',
+					},
+					amount: {
+						type: 'string',
+						format: 'int64',
+					},
+				},
+			},
+		},
+	},
+};
+
+const SIZE_INT64 = 8;
+const SIZE_UINT64 = SIZE_INT64;
+// tslint:disable-next-line no-magic-numbers
+const TEN_UNIT = BigInt(10) * BigInt(10) ** BigInt(8);
+const MAX_VOTE = 10;
+const MAX_UNLOCKING = 20;
+
+export interface RawAssetVote {
+	readonly delegateAddress: string;
+	readonly amount: string;
+}
+
+interface RawAsset {
+	readonly votes: ReadonlyArray<RawAssetVote>;
+}
+
+export class VoteTransaction extends BaseTransaction {
+	public readonly asset: VoteAsset;
+	public static TYPE = 13;
+
+	public constructor(rawTransaction: unknown) {
+		super(rawTransaction);
+		const tx = (typeof rawTransaction === 'object' && rawTransaction !== null
+			? rawTransaction
+			: {}) as Partial<TransactionJSON>;
+		if (tx.asset) {
+			const rawAsset = tx.asset as RawAsset;
+			this.asset = {
+				votes: rawAsset.votes.map(vote => {
+					const amount = isNumberString(vote.amount)
```

Comment:
`amount` is marked as required in the schema then do we need this ternary operation? Can we just always cast it to BigInt?

Reply:
This constructor is used before validation, so it should be checked for not throwing